assets/js/tx-sender.js

```diff
@@ -0,0 +1,465 @@
+    resp = await submit_and_verify({
+      TransactionType: "OfferCreate",
+      Account: sending_address,
+      TakerGets: "1000000000000000", // 1 billion XRP
+      TakerPays: {
+        currency: pp_sending_currency,
+        value: "1000000000",
+        issuer: pp_issuer_address
+      }
+    }, sending_secret, true)
+    if (resp === undefined) {
+      console.log("Error placing order to enable partial payments")
+      return
+    }
+    $("#pp_progress .progress-bar").width("100%").removeClass("progress-bar-animated")
+    $("#pp_progress").hide()
+
+    // Done. Enable "Send Partial Payment" button
+    console.log("Done getting ready to send partial payments.")
+    $("#send_partial_payment button").prop("disabled",false)
+    $("#send_partial_payment button").attr("title", "")
+  }
+  set_up_for_partial_payments()
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Button Handlers
+  //////////////////////////////////////////////////////////////////////////////
+
+  // 1. Send XRP Payment Handler -------------------------------------------
+  async function on_click_send_xrp_payment(event) {
+    const destination_address = $("#destination_address").val()
+    const xrp_drops_input = $("#send_xrp_payment_amount").val()
+    $("#send_xrp_payment .loader").show()
+    $("#send_xrp_payment button").attr("disabled","disabled")
+    await submit_and_verify({
+      TransactionType: "Payment",
+      Account: sending_address,
+      Destination: destination_address,
+      Amount: xrp_drops_input
+    })
+    $("#send_xrp_payment .loader").hide()
+    $("#send_xrp_payment button").attr("disabled",false)
+
+  }
+  $("#send_xrp_payment button").click(on_click_send_xrp_payment)
+
+  // 2. Send Partial Payment Handler ---------------------------------------
+  async function on_click_send_partial_payment(event) {
+    const destination_address = $("#destination_address").val()
+    $("#send_partial_payment .loader").show()
+    $("#send_partial_payment button").attr("disabled","disabled")
+
+    // const path_find_result = await api.request("ripple_path_find", {
+    //   source_account: sending_address,
+    //   destination_account: destination_address,
+    //   destination_amount: "-1", // as much XRP as possible
+    //   source_currencies: [{currency: pp_sending_currency, issuer: pp_issuer_address}]
+    // })
+    // console.log("Path find result:", path_find_result)
+    // use_path = path_find_result.alternatives[0].paths_computed
+
+    await submit_and_verify({
+      TransactionType: "Payment",
+      Account: sending_address,
+      Destination: destination_address,
+      Amount: "1000000000000000", // 1 billion XRP
+      SendMax: {
+        value: (Math.random()*.01).toPrecision(15), // random very small amount
+        currency: pp_sending_currency,
+        issuer: pp_issuer_address
+      },
+      Flags: 0x80020000 // tfPartialPayment | tfFullyCanonicalSig
```

Comment:
I think you can actually do:
```js
      Flags: api.txFlags.Payment.PartialPayment | api.txFlags.Universal.FullyCanonicalSig
```

Actually, prepareTransaction always sets FullyCanonicalSig for you, so this should work as well:
```js
      Flags: api.txFlags.Payment.PartialPayment
```

Reply:
That's good to know, but I think it's still a good practice to set FullyCanonicalSig explicitly since there are APIs and situations, esp. with multisigning, where it won't be enabled automatically. So the first example seems better.